### ISSUE TITLE
[feature request] Hints on un-implemented virtual methods

### ISSUE THREAD

**Issue by FlyinCow** [clangd/clangd#2240]
A common build error is programmers forgetting to implement pure virtual methods inherit from some abstract base class. Static analyse should be able to identify this situation.

Enhancements:
- add refactor option for class: `implement virtual methods for ...`
- [optional] mark not implemented pure virtual methods in base class, provide link to subclass
- mark errors on subclass that has unimplemented pure virtual methods, provide link to method declaration in base class

**Comment by HighCommander4** [comment]
Transferred to server repository as these are all features that need implementation in the clangd server.

I believe the first bullet point is covered by https://github.com/clangd/clangd/issues/1037.

**Comment by FlyinCow** [comment]
> Transferred to server repository as these are all features that need implementation in the clangd server.
> 
> I believe the first bullet point is covered by #1037.

Yes it covers but I can't trigger it in my vscode, are there some flags in settings or some arguments?

**Comment by HighCommander4** [comment]
> > Transferred to server repository as these are all features that need implementation in the clangd server.
> > I believe the first bullet point is covered by #1037.
> 
> Yes it covers but I can't trigger it in my vscode, are there some flags in settings or some arguments?

I just meant there is also an issue on file about it.

The issue is open, the feature has not been implemented.

**Comment by FlyinCow** [comment]
> Note that `clang` already implements some diagnostics for this.
> 
> ![Image](https://github.com/user-attachments/assets/a4bdff0c-2daf-4688-ae14-54d6cf76a3a1)

Yes, and since it can be detected it would be nice if such code could be generated by some code actions.
